libdevcore/CommonData.cpp

```diff
@@ -181,3 +182,15 @@ bool dev::isValidDecimal(string const& _string)
 		return false;
 	return true;
 }
+
+// Returns a quoted string if all characters are printable ASCII chars,
+// or its hex representation otherwise.
+std::string dev::formatAsStringOrNumber(std::string const& _value)
+{
+	if (_value.length() <= 32)
```

Comment:
So if it is not printable as text, but is longer than 32 bytes, it is still printed as unprintable text?

Reply:
I'll change it to an assertion. String literals can be at most 32 bytes in yul.